misc/python/materialize/git.py

```diff
@@ -238,6 +282,30 @@ def get_commit_message(commit_sha: str) -> str | None:
         return None
 
 
+def get_branch_name() -> str:
+    command = ["git", "branch", "--show-current"]
+    return spawn.capture(command).strip()
+
+
+def get_previous_version(version: MzVersion) -> Version:
+    if version.prerelease is not None and len(version.prerelease) > 0:
+        # simply drop the prerelease, do not try to find a decremented version
+        return MzVersion.create_mz(version.major, version.minor, version.patch)
+
+    # type must match for comparison
+    version_as_semver_version = version.to_semver()
```

Comment:
I'm a bit confused by this, isn't MzVersion also a Version?

Reply:
It is but the equals method will scream. It checks that the type matches.

```
    mzv = MzVersion.parse_mz("v1.10.1")
    v = mzv.to_semver()
    print(mzv == v)
```

results in 

```
TypeError: Expected str, bytes, dict, tuple, list, or MzVersion instance, but got <class 'semver.version.Version'>
```

Related (but Java): https://stackoverflow.com/a/596766/584532

Reply:
I think we should get rid of all usages of `Version` in the code if possible with reasonable effort. It's also dangerous in other places, e.g., if you want to have the string representation which will differ depending on what instance you have.